libraries/dotnet/EnergyOrigin.IntegrationEvents/EnergyOrigin.IntegrationEvents/Events/IntegrationEvent.cs

```diff
@@ -0,0 +1,15 @@
+namespace EnergyOrigin.IntegrationEvents.Events;
+
+public abstract record IntegrationEvent
+{
+    protected IntegrationEvent(Guid id, Guid traceId, DateTimeOffset created)
+    {
+        Id = id;
```

Comment:
Why is Id needed when you have traceid?

Reply:
TraceId is for the trace. Id is so that we can uniquely identify the event. If we will ever need it, I dunno, but an architect decided this